lambdas/address/src/main/java/uk/gov/di/ipv/cri/address/api/handler/AddressDateLinker.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.di.ipv.cri.address.api.handler;
+
+import uk.gov.di.ipv.cri.address.library.exception.AddressProcessingException;
+import uk.gov.di.ipv.cri.common.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.cri.common.library.persistence.item.CanonicalAddress;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+
+public class AddressDateLinker {
+
+    @ExcludeFromGeneratedCoverageReport
+    private AddressDateLinker() {
+        throw new IllegalStateException("Static Utility class");
+    }
+
+    public static void linkCurrentAndPreviousAddressesDates(List<CanonicalAddress> addresses)
+            throws AddressProcessingException {
+
+        if (addresses.size() > 2) {
+            // A non-null validFrom only exists in the first (CURRENT) address
+            throw new AddressProcessingException("Too many Addresses.");
+        }
+
+        CanonicalAddress prevProcessedAddress = null;
+
+        for (CanonicalAddress address : addresses) {
```

Comment:
There is no guarantee this list will always be in the correct order. We would be better reading the validFroms and ordering the list ourselves then making the preceding validFrom of each the valid until of the next

Reply:
Linker has be updated to not assume the order of addresses and add additional address validation.